libfwupdplugin/fu-bluez-device.c

```diff
@@ -171,24 +171,19 @@ fu_bluez_device_set_modalias(FuBluezDevice *self, const gchar *modalias)
 		fu_firmware_strparse_uint16_safe(modalias, modaliaslen, 21, &rev, NULL);
 	}
 
-	if (vid != 0x0 && pid != 0x0 && rev != 0x0) {
-		g_autofree gchar *devid = NULL;
-		devid = g_strdup_printf("BLUETOOTH\\VID_%04X&PID_%04X&REV_%04X", vid, pid, rev);
-		fu_device_add_instance_id(FU_DEVICE(self), devid);
-	}
-	if (vid != 0x0 && pid != 0x0) {
-		g_autofree gchar *devid = NULL;
-		devid = g_strdup_printf("BLUETOOTH\\VID_%04X&PID_%04X", vid, pid);
-		fu_device_add_instance_id(FU_DEVICE(self), devid);
-	}
+	/* add generated IDs */
+	if (vid != 0x0)
+		fu_device_instance_assign_uint16(FU_DEVICE(self), "VID", vid);
```

Comment:
@superm1 I did ponder calling this fu_device_instance_assign_u16 to make it shorter, but we use uint16 in other parts of the code.

Reply:
Maybe it makes sense to go the shorter approach and drop a comment for the next time we break ABI to rename the others?

Reply:
Makes sense to me. The other thing I pondered was to rename `fu_device_instance_assign_safe` to `fu_device_instance_assign` and  `fu_device_instance_assign` to  `fu_device_instance_assign_unsafe` just so it's obvious we're using the unsafe one. Or maybe the latter should be  `fu_device_instance_assign_raw` to show we're not going to be substituting spaces and such.

Reply:
Also, maybe `fu_device_instance_assign` to `fu_device_instance_add` might be better as well and more in keeping? Ideas welcome.

Reply:
function names entirely up to you here, I don't feel strongly.